st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -102,6 +103,8 @@ def _handle_schedule_execution(self, liveaction):
         except jsonschema.ValidationError as e:
             LOG.exception('Unable to execute action. Parameter validation failed.')
             abort(http_client.BAD_REQUEST, re.sub("u'([^']*)'", r"'\1'", e.message))
+        except UniqueTraceNotFoundException as e:
+            abort(http_client.BAD_REQUEST, str(e))
```

Comment:
I would handle the StackStormDBObjectNotFoundError exception here rather than this one.